logstash-core/lib/logstash/config/source/multi_local.rb

```diff
@@ -23,14 +23,17 @@ class MultiLocal < Local
     include LogStash::Util::SubstitutionVariables
     include LogStash::Util::Loggable
 
+    REMOVE_COMMENTS_CONFIG_KEYS = %w(config.string)
```

Comment:
review note: currently only `config.string` may contain the comments. `YAML::safe_load` removes the comments from other keys such as `pipeline.id`.